ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/timeline/TimelineComponentView.kt

```diff
@@ -82,39 +92,47 @@ internal fun TimelineComponentView(
                 modifier = Modifier.height(timelineState.itemSpacing.dp)
                     .constrainAs(itemSpacingRef) {
                         top.linkTo(bottomContentBarrier)
-                        if (isLastItem) {
-                            bottom.linkTo(parent.bottom)
-                        }
                     },
             )
 
-            IconComponentView(
-                style = item.icon,
-                state = state,
+            Box(
                 modifier = Modifier.constrainAs(iconRef) {
                     when (timelineState.iconAlignment) {
                         TimelineComponent.IconAlignment.Title -> {
-                            top.linkTo(titleRef.top)
-                            bottom.linkTo(titleRef.bottom)
+                            top.linkTo(currentPreviousItem ?: parent.top)
```

Comment:
This is part of the fixes I applied to make sure the rows didn't overlap each other.